api/app/analysis/natural_lands/analysis.py

```diff
@@ -0,0 +1,191 @@
+import json
+import logging
+import os
+import traceback
+from functools import partial
+
+import dask.dataframe as dd
+import duckdb
+import numpy as np
+import pandas as pd
+import s3fs
+from flox.xarray import xarray_reduce
+
+from dask.distributed import Client, LocalCluster
+from pathlib import Path
+import asyncio
+
+from ..common.analysis import (
+    get_geojson,
+    read_zarr_clipped_to_geojson,
+)
+from .query import create_gadm_natural_lands_query
+
+NATURAL_LANDS_CLASSES = {
+    2: "Natural forests",
+    3: "Natural short vegetation",
+    4: "Natural water",
+    5: "Mangroves",
+    6: "Bare",
+    7: "Snow",
+    8: "Wetland natural forests",
+    9: "Natural peat forests",
+    10: "Wetland natural short vegetation",
+    11: "Natural peat short vegetation",
+    12: "Cropland",
+    13: "Built-up",
+    14: "Non-natural tree cover",
+    15: "Non-natural short vegetation",
+    16: "Non-natural water",
+    17: "Wetland non-natural tree cover",
+    18: "Non-natural peat tree cover",
+    19: "Wetland non-natural short vegetation",
+    20: "Non-natural peat short vegetation",
+    21: "Non-natural bare",
+}
+
+
+async def zonal_statistics_on_aois(aois, dask_client):
+    geojsons = await get_geojson(aois)
+    aois = sorted(
+        [{"type": aois["type"], "id": id} for id in aois["ids"]],
+        key=lambda aoi: aoi["id"],
+    )
+
+    precompute_partial = partial(zonal_statistics)
+    dd_df_futures = await dask_client.gather(
+        dask_client.map(precompute_partial, aois, geojsons)
+    )
+    dfs = await dask_client.gather(dd_df_futures)
+    df = await dask_client.compute(dd.concat(dfs))
+
+    return df
+
+
+async def zonal_statistics(aoi, geojson):
+    natural_lands_obj_name = "s3://gfw-data-lake/sbtn_natural_lands/zarr/sbtn_natural_lands_all_classes.zarr"
+    pixel_area_obj_name = "s3://gfw-data-lake/umd_area_2013/v1.10/raster/epsg-4326/zarr/pixel_area.zarr/"
+    natural_lands = read_zarr_clipped_to_geojson(natural_lands_obj_name, geojson).band_data
+    natural_lands.name = "natural_lands_class"
+    pixel_area = read_zarr_clipped_to_geojson(pixel_area_obj_name, geojson)
+
+    groupby_layers = [natural_lands]
+    expected_groups = [np.arange(1, 22)]
+
+    counts = xarray_reduce(
+        pixel_area,
+        *tuple(groupby_layers),
+        func="count",
```

Comment:
func  should be "sum"

Reply:
May be worth adding unit tests that would catch this, like this one https://github.com/wri/project-zeno-data-infra/blob/989cb39b5f6430f4a7045742aa6c7cb8663d6e75/api/test/unit/test_grasslands_analysis.py#L115.